examples/text-generation/run_generation.py

```diff
@@ -313,8 +321,164 @@ def main():
         use_lazy_mode = False
 
     import habana_frameworks.torch.hpu as torch_hpu
+    if args.dataset_name == "openorca":
+        # Benchmark over the prompts below
+        def get_ds(args):
+            ds = pd.read_pickle(args.dataset)
+            return ds
+
+
+        def get_input(ds, batch_size):
+            queries = []
+            tok_input = ds["tok_input"].tolist()
+            for start in range(0, len(ds), batch_size):
+                end = start + batch_size
+                batch = tok_input[start:end]
+                input_ids = []
+                attention_mask=[]
+                for query in batch:
+                    input_ids.append(
+                        [0] * (args.max_input_tokens - len(query)) + query)
+                    attention_mask.append([0] * (args.max_input_tokens - len(query)) + [1] * len(query))
+                queries.append({
+                    'input_ids': torch.tensor(input_ids, dtype=torch.int32),
+                    'attention_mask': torch.tensor(attention_mask, dtype=torch.int32)
+                })
+            return queries
+
+        ds = get_ds(args)
+        input_sentences = get_input(ds, args.batch_size)
+
+        def generate(input_tokens, size=None, reduce_recompile=False):
+            """Generates sequences from the input sentences and returns them."""
+
+            t0 = time.perf_counter()
```

Comment:
please check how the 1st token latency is done here, and do the same https://github.com/huggingface/optimum-habana/blob/main/examples/text-generation/run_generation.py#L381

Reply:
please note this for evaluation and not for latency test still existing behaviour is retained